proposals/dao/merger.ts

```diff
@@ -0,0 +1,116 @@
+  }
+
+  // 1. Deploy TribeRariDAO
+  const factory = await ethers.getContractFactory('FeiDAO');
+  const tribeRariDAO = await factory.deploy(tribe, rariTimelock, ethers.constants.AddressZero);
+
+  await tribeRariDAO.deployTransaction.wait();
+
+  logging && console.log('tribeRariDAO: ', tribeRariDAO.address);
+
+  // 2. Deploy PegExchanger
+  const pegFactory = await ethers.getContractFactory('PegExchanger');
+  const pegExchanger = await pegFactory.deploy(tribeRariDAO.address);
+
+  await pegExchanger.deployTransaction.wait();
+
+  logging && console.log('pegExchanger: ', pegExchanger.address);
+
+  // 3. Deploy TribeRagequit
+  const ragequitFactory = await ethers.getContractFactory('TRIBERagequit');
+  const tribeRagequit = await ragequitFactory.deploy(merkleRoot, rageQuitDeadline, tribeRariDAO.address);
+
+  await tribeRagequit.deployTransaction.wait();
+
+  logging && console.log('tribeRagequit: ', tribeRagequit.address);
+
+  return {
+    tribeRariDAO,
+    pegExchanger,
+    tribeRagequit
+  } as NamedContracts;
+};
+
+export const setup: SetupUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  logging && console.log('Setup');
+  const guardian = addresses.multisig;
+  const tribeRagequit: TRIBERagequit = contracts.tribeRagequit as TRIBERagequit;
+
+  const signer = await getImpersonatedSigner(guardian);
+
+  await tribeRagequit.connect(signer).setExchangeRate(equity);
+};
+
+export const teardown: TeardownUpgradeFunc = async (addresses, oldContracts, contracts, logging) => {
+  logging && console.log('Teardown');
+  const proposal = await constructProposal(
+    rariMergerProposal,
+    contracts as unknown as MainnetContracts,
+    addresses,
+    logging
+  );
+
+  const timelock: Timelock = (await contracts.rariTimelock) as Timelock;
+
+  await proposal.setGovernor(await timelock.admin());
+
+  logging && console.log(`Simulating proposal...`);
+  await proposal.simulate();
+
+  const tribeRagequit: TRIBERagequit = contracts.tribeRagequit as TRIBERagequit;
+  const pegExchanger: PegExchanger = contracts.pegExchanger as PegExchanger;
+  await tribeRagequit.setTribeRariDAOActive();
+  await pegExchanger.setTribeRariDAOActive();
+};
+
+export const validate: ValidateUpgradeFunc = async (addresses, oldContracts, contracts) => {
+  const tribeRagequit: TRIBERagequit = contracts.tribeRagequit as TRIBERagequit;
+  const pegExchanger: PegExchanger = contracts.pegExchanger as PegExchanger;
+  const { tribe, fei } = contracts;
+
+  expect(await tribeRagequit.isEnabled()).to.be.true;
+  expect(await pegExchanger.isEnabled()).to.be.true;
+
+  expect((await tribeRagequit.intrinsicValueExchangeRateBase()).toString()).to.be.equal('1237113801');
+  expect((await tribe.balanceOf(pegExchanger.address)).toString()).to.be.equal('270000000000000000000000000');
+  expect((await fei.balanceOf(addresses.gfxAddress)).toString()).to.be.equal('315909060000000000000000');
```

Comment:
ragequit deadline as well as merkle root should be checked in this validate function